content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -1,48 +1,75 @@
-We will be using the Client\Server architecture to illustrate more on sockets. Client and server communicate by writing to and reading from the socket connection. To establish this connection, the machines must have each other's network information.
+
+Socket programming is a means of communicating data between two computers across a network. Connections can be made using either a **connection-oriented protocol** or a **connectionless protocol**. TCP/IP, a connection-oriented protocol, will be used in our case.
+
+Before exchanging data, two computers must establish a link in connection-oriented protocols. In contrast, you don't require every packet to arrive at its intended destination for connectionless. It is made feasible via UDP (User Datagram Protocol). To demonstrate sockets further, we shall use the Client\Server architecture. Client and server communicate by writing to and reading from the socket connection. The machines must have each other's network information to make this connection.
+
 Table of contents:
 - [Definition-of-a-Socket](#definition-of-a-socket)
-- [Server-Side-Programming ](#server-side-programming)
-- [Client-Side-Programming](#client-side-programming)
-- [How-to-Run-the-Programs-on-different-applications](#how-to-run-the-programs-on-diffrent-applications)
+- [Programming-on-the-Server-side ](#programming-on-the-server-side)
+- [Programming-on-the-Client-side](#programming-on-client-side)
+- [How-to-Run-Programs-in-a-Variety-of-Applications](#how-to-run-programs-in-a-variety-of-applications)
 - [Conclusion](#conclusion)
+
 ### Definition of a Socket
-A socket is an **endpoint** of communication, acting as a connection between two computers on a network. It contains a socket number that the TCP layer can use to identify the application to which data will be transmitted. Additionally, an endpoint is made up of a port number and an IP address.
-There are two types of sockets
-1. Server socket - which waits for a request from clients
-2. Client socket - can be used to send and receive a request from clients.
-The client must know two information about the server:
-1.IP address of Server and 
+
+A Socket is a communication **endpoint** that serves as a link between two machines on a network. It has a socket number, which the TCP/IP layer can use to identify the application that receives the data. Additionally, a port number and an IP address are what make up an endpoint.
+
+Transmission Control Protocol (TCP) - This is one of the most important protocols of the Internet Protocols suite. It is a widely used protocol for data transmission in a network that operates in Client/Server point-to-point mode.
+
+Two categories of Sockets:
+
+1. A server socket awaits a request from a client.
+2. Client socket - utilized for sending and receiving requests from clients.
+
+The client has to know two things about the server:
+
+1.Server's IP address 
```

Comment:
> 1.Server's IP address 
> 2.Port number

Add a space after the numbers and rephrase it to something like this:

> 1. Its IP address 
> 2. Its port number

I rephrased it like this to be more grammatically correct and follow the proceeding sentence.